### PR TITLE
[VL] Bugfix for VeloxMemoryManager's constructor

### PR DESCRIPTION
**Pull request by zuochunwei**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

(Fixes: \#ISSUE-ID)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Review by zhztheplayer — APPROVED** [review]
Thanks!

File: cpp/velox/memory/VeloxMemoryManager.h [hunk 1]
```diff
@@ -44,13 +44,22 @@ class VeloxMemoryManager final : public MemoryManager {
...
  private:
+  facebook::velox::memory::IMemoryManager::Options getOptions(
+      std::shared_ptr<MemoryAllocator> allocator,
+      std::shared_ptr<AllocationListener> listener) const;
+
   std::string name_;
 
+#ifdef GLUTEN_ENABLE_HBM
+  std::unique_ptr<VeloxMemoryAllocator> wrappedAlloc_;
+#endif
```
Comment (zhztheplayer) [inline cpp/velox/memory/VeloxMemoryManager.h:55]: Member order: This better be placed after `glutenAlloc_` ?

Comment (zuochunwei) [reply cpp/velox/memory/VeloxMemoryManager.h:55]: seems to be no relation between these two data members

File: cpp/velox/memory/VeloxMemoryManager.cc [hunk 2]
```diff
@@ -162,41 +162,52 @@ class ListenableArbitrator : public velox::memory::MemoryArbitrator {
   gluten::AllocationListener* listener_;
 };
 
-VeloxMemoryManager::VeloxMemoryManager(
-    std::string name,
+velox::memory::IMemoryManager::Options VeloxMemoryManager::getOptions(
     std::shared_ptr<MemoryAllocator> allocator,
-    std::shared_ptr<AllocationListener> listener)
-    : MemoryManager(), name_(name), listener_(std::move(listener)) {
+    std::shared_ptr<AllocationListener> listener) const {
```
Comment (Yohahaha) [inline cpp/velox/memory/VeloxMemoryManager.cc:167]: If use `listener_` to construct ListenableArbitrator, why pass a unused `listener`?

**Comment by Yohahaha** [comment]
Request review or at least a notification to author before modify the codes you just read is a good way, and if you like and will do code refactor, then do it more carefully please.

This feature was not complete and lots follow up patches were preparing, just wait codes become more stable before refactor.

**Comment by zhztheplayer** [comment]
This feature was not complete and lots follow up patches were preparing, just wait codes become more stable before refactor.

--
The PR was to fix a serious dangling pointer bug. @zuochunwei Would you like to add some descriptions to the PR title about that bug?

**Comment by zhztheplayer** [comment]
> Request review or at least a notification to author

I agree with that. However it's hard to define "author" for every part of our code since most of them were based on collaborations. A healthy OSS community doesn't always have authors (although it probably authored by a couple of individuals). Do you agree with that?

> This feature was not complete and lots follow up patches were preparing

Let's open an umbrella issue to organize the pending PRs (about the memory manager)? I realized that you had contributed a lot of patches these days and many thanks to that. However more PRs you raised, more chances they get conflict during development iteration of the main code base. I'm sorry if a merge causes conflicts for your PRs and I'll track the umbrella issue to have a initial check before merging other PRs.

**Pull request merged**